src/main/java/views/add_edit_category_views/EditCategoryV.java

```diff
@@ -0,0 +1,95 @@
+    JTextField nameInput;
+    JTextField budgetInput;
+    String selectedCategory;
+    int monthID;
+    SessionStorage currSession;
+
+    public EditCategoryV(CategoryC controller, String[] existingCategory, int monthID, SessionStorage currSession) {
+        /*
+          Builds EditCategoryV.
+         */
+        JLabel select_category_label = new JLabel(" Select existing category:");
+        this.categoryCombo = new JComboBox<>(existingCategory); // category list
+        JLabel name_label = new JLabel("New Category Name:");
+        this.nameInput = new JTextField(15);
+        JLabel budget_label = new JLabel(" New Category Budget:");
+        this.budgetInput = new JTextField(15);
+        JButton submit = new JButton("Submit");
+        submit.setSize(30,10);
+
+        JFrame frame = new JFrame();
+        JPanel panel = new JPanel();
+        panel.setBorder(BorderFactory.createEmptyBorder(50, 30, 50, 30));
+        panel.setLayout(new GridLayout(0,1));
+        JPanel panell = new JPanel();
+        panell.setBorder(BorderFactory.createEmptyBorder(50, 30, 50, 30));
+        panell.setLayout(new GridLayout(0,1));
+
+        frame.add(panel, BorderLayout.NORTH);
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setTitle("Edit Category");
+        frame.setSize(300,500);
+        frame.setSize(500,300);
+        panel.add(select_category_label);
+        panel.add(categoryCombo);
+        panel.add(name_label, BorderLayout.WEST);
+        panel.add(nameInput, BorderLayout.CENTER);
+        panel.add(budget_label);
+        panel.add(budgetInput);
+        frame.add(panell, BorderLayout.SOUTH);
+        panell.add(submit);
+
+        frame.pack();
+        frame.setVisible(true);
+
+        submit.addActionListener(this);
+        categoryCombo.addActionListener(this);
+
+        this.controller = controller;
+        this.monthID = monthID;
+        this.currSession = currSession;
+    }
+    @Override
+    public void actionPerformed(ActionEvent evt) {
+        /*
+          Two ActionListeners with different behaviours differentiated by evt.getSource().
+          Formats user input to pass in valid parameters for a CategtoryC to start a use case.
+          Pop-up window with context specific message may be shown to user.
+         */
+        if (evt.getSource() == categoryCombo) {
+            this.selectedCategory = (String) categoryCombo.getSelectedItem();
+        }
+        else {
+            CategoryOD message = null;
+            try {
+                message = controller.categoryInMonth(nameInput.getText(), String.valueOf(budgetInput), monthID, currSession, selectedCategory);
+            } catch (EntityException e) {
+                JOptionPane.showMessageDialog( this, "This month does not exist in current session. Please go to add month page.");
+            }
+            if(message != null){
+                JOptionPane.showMessageDialog( this, message.getMessage());}
```

Comment:
Could you try something like this?

```suggestion
            try {
                CategoryOD message = controller.categoryInMonth(nameInput.getText(), String.valueOf(budgetInput), monthID, currSession, selectedCategory);
                JOptionPane.showMessageDialog( this, message.getMessage());
            } catch (EntityException e) {
                JOptionPane.showMessageDialog( this, "This month does not exist in current session. Please go to add month page.");
            }
```

Reply:
The point of message = null is to check if a use case was run. This is because I'm trying to catch the EntityException thrown by an invalidmonthID (that is not related to my feature so I figured that it makes more sense to be handled here). I'm also adding some if statements to make sure user inputs are complete, also makes more sense to be handled before entering a use case. It should be more clear in my next commit!